pkg/util/kubeobjects/topology/constraint.go

```diff
@@ -1,13 +1,12 @@
-package utils
+package topology
 
 import (
 	"github.com/Dynatrace/dynatrace-operator/pkg/util/kubeobjects/labels"
-	appsv1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-func BuildTopologySpreadConstraints(topologySpreadConstraints []corev1.TopologySpreadConstraint, appLabels *labels.AppLabels) []corev1.TopologySpreadConstraint {
+func SpreadConstraints(topologySpreadConstraints []corev1.TopologySpreadConstraint, appLabels *labels.AppLabels) []corev1.TopologySpreadConstraint {
```

Comment:
you moved it to a very general place
now the name of it is misleading

It is a "use provided or set defaults"

I would:
- move the `if len(topologySpreadConstraints) > 0` check where you use this
  - having something to overwrite it is a component specific things, you are in a non-component-specific package
- have the rest remain here and rename the func to:
```
func MaxOnePerNode(appLabels *labels.AppLabels) []corev1.TopologySpreadConstraint {
return []corev1.TopologySpreadConstraint{
			{
				MaxSkew:           1,
				TopologyKey:       "topology.kubernetes.io/zone",
				WhenUnsatisfiable: "ScheduleAnyway",
				LabelSelector:     &metav1.LabelSelector{MatchLabels: appLabels.BuildMatchLabels()},
			},
			{
				MaxSkew:           1,
				TopologyKey:       "kubernetes.io/hostname",
				WhenUnsatisfiable: "DoNotSchedule",
				LabelSelector:     &metav1.LabelSelector{MatchLabels: appLabels.BuildMatchLabels()},
			},
		}
}
```

Reply:
fixed in 1e7389728